apps/bundle-analyzer/components/sidebar.tsx

```diff
@@ -76,101 +117,136 @@ export function Sidebar({
-      style={{ width: `${sidebarWidth}%` }}
-    >
-      <div className="flex-1 p-3 space-y-8 overflow-y-auto">
-        <div className="space-y-2">
-          <h2 className="text-s font-semibold mb-1 text-foreground truncate">
-            {selectedSource
-              ? selectedSource.path || 'All Route Modules'
-              : 'Unknown Source'}
-          </h2>
-          {selectedSourceIndex != null &&
-          analyzeData.source(selectedSourceIndex) ? (
-            <div>
-              <div className="text-xs">
-                <span>
-                  {hasChildModules
-                    ? formatBytes(
-                        analyzeData.getSourceRecursiveSize(selectedSourceIndex)
-                      )
-                    : formatBytes(
-                        analyzeData.getSourceOutputSize(selectedSourceIndex)
-                      )}
-                </span>{' '}
-                <span className="text-muted-foreground">bundled</span>
-              </div>
-              {hasChildModules && childModuleCount != null ? (
-                <div className="text-xs">
-                  <span>{childModuleCount} </span>
-                  <span className="text-muted-foreground">
-                    {childModuleCount === 1 ? 'module' : 'modules'}
-                  </span>
-                </div>
-              ) : null}
-            </div>
-          ) : null}
-        </div>
+  const { size, compressedSize } = analyzeData.getRecursiveSizes(
+    selectedSourceIndex,
+    filterSource
+  )
 
-        {selectedSourceIndex != null &&
-          analyzeData.source(selectedSourceIndex) &&
-          (specialModuleType === SpecialModule.POLYFILL_MODULE ||
-            specialModuleType === SpecialModule.POLYFILL_NOMODULE) && (
-            <dl>
-              <div className="flex items-center gap-2">
-                <dt className="inline-flex items-center">
-                  <Badge variant="polyfill">Polyfill</Badge>
-                </dt>
-                <dd className="text-xs text-muted-foreground">
-                  Next.js built-in polyfills
-                </dd>
-              </div>
-            </dl>
-          )}
+  const chunks =
+    selectedSourceIndex != null
+      ? analyzeData.sourceChunks(selectedSourceIndex)
+      : []
 
+  return (
+    <div className="flex-1 p-3 space-y-8 overflow-y-auto">
+      <div className="space-y-2">
+        <h2 className="text-s font-semibold mb-1 text-foreground truncate">
+          {selectedSource?.path || 'All Route Modules'}
+        </h2>
         {selectedSourceIndex != null &&
-          analyzeData.source(selectedSourceIndex) &&
-          !hasChildModules && (
+        analyzeData.source(selectedSourceIndex) ? (
+          <div className="text-xs">
+            <div>
+              <span>{formatBytes(size)}</span>{' '}
+              <span className="text-muted-foreground">uncompressed</span>
+              <TooltipProvider>
```

Comment:
This boilerplate could be shared between the two tooltips by hoisting into a component at the top of the module that accepts a `children` prop for the message.